openrouteservice/src/main/java/org/heigit/ors/api/requests/matrix/MatrixRequestHandler.java

```diff
@@ -81,9 +92,59 @@ public static MatrixResult generateMatrixFromRequest(MatrixRequest request) thro
         if (request.hasUnits())
             coreRequest.setUnits(convertUnits(request.getUnits()));
 
+        MatrixSearchParameters params = new MatrixSearchParameters();
+        if(request.hasMatrixOptions())
+            coreRequest.setFlexibleMode(processMatrixRequestOptions(request, params));
+        coreRequest.setSearchParameters(params);
         return coreRequest;
     }
 
+    private static boolean processMatrixRequestOptions(MatrixRequest request, MatrixSearchParameters params) throws StatusCodeException {
+        MatrixRequestOptions routeOptions = request.getMatrixOptions();
+        try {
+            int profileType = convertRouteProfileType(request.getProfile());
+            params.setProfileType(profileType);
+        } catch (Exception e) {
+            throw new ParameterValueException(RoutingErrorCodes.INVALID_PARAMETER_VALUE, RouteRequest.PARAM_PROFILE);
+        }
+        boolean flexibleMode = processRequestOptions(routeOptions, params);
+        return flexibleMode;
+    }
+
+    public static int convertRouteProfileType(APIEnums.Profile profile) {
```

Comment:
It appears to me that there is massive code duplication across `MatrixRequestHandler`, `RouteRequestHandler` and `GenericRequestHandler`. For example, the `convertRouteProfileType` method here is the same as the corresponding one in the other two classes. I think it would make sense to refactor things to have a common upstream method implementation in `GenericRequestHandler` from which this class could inherit. The same applies for a lot of other methods from this class such as `convertAvoidAreas`, `convertAvoidCountries`, `convertFeatureTypes`, etc.

Reply:
@takb Is this planned?

Reply:
@takb Can you please give me some insight on whether this is planned in the coming API/Core updates?

Reply:
Honestly, I do not fell entirely comfortable with approving changes which introduce copies of methods from other classes without a clear plan on addressing this afterwards. Before we move on, could you maybe create a separate issue describing the problem such that we don't loose it from our radar? Cheers!

Reply:
The problem is that I need to fix API code in order to write a new algorithm. There is a defined project beforehand and if the circumstances for completing it are not given, I have to find a solution, which I did here. If you take a look at the core channel, I have indeed tried to get insights into this during development, but there is no progress on this. I have deadlines for this project and I am not going to go over them for reasons that are clearly out of my scope.
I will create an issue for this, but this is not my issue.

Reply:
Here you go.
https://github.com/GIScience/openrouteservice/issues/1067

Reply:
Thanks a lot! You are of course right, fixes to existing codebase are clearly not within the scope of this project. Still,  implementing new functionality provides the opportunity of stepping outside of the project boundaries and choosing solutions which lead to a reduction of issues in broader picture rather than aggravating them. But as this is voluntary I cannot argue with your choices.

Reply:
Sure, it's nice to make the codebase nice while working on an algorithm, given you have the time. Which I do not, because there is a deadline.